vowpalwabbit/learner.cc

```diff
@@ -73,6 +73,10 @@ inline bool example_is_newline_not_header(example& ec, vw& all)
   // If we are using CCB, test against CCB implementation otherwise fallback to previous behavior.
   bool is_header = false;
   if (all.example_parser->lbl_parser.label_type == label_type_t::ccb) { is_header = CCB::ec_is_example_header(ec); }
+  else if (all.example_parser->lbl_parser.label_type == label_type_t::cs)
+  {
+    is_header = COST_SENSITIVE::ec_is_example_header(ec);
+  }
   else
   {
     is_header = CB::ec_is_example_header(ec);
```

Comment:
Does it make sense to have this be a function, especially since you duplicate the logic in shared_feature_merger.cc?

There are a few other place we check for headers, too, and I am unsure whether they are doing it correctly in all situations:

* https://github.com/VowpalWabbit/vowpal_wabbit/blob/6335dd190d682bd98a01999a4b5b091ea69e6dae/vowpalwabbit/explore_eval.cc#L66
* https://github.com/VowpalWabbit/vowpal_wabbit/blob/21030f5c97b71f91224bfb44c4381316cf82efeb/vowpalwabbit/search.cc#L1277

(There are other places in search). If it is not clear what the right behaviour is for search, do not let it block this fix, though if you could create an issue tracking checking whether the is_example_header is properly checked in search assigned to me that would be appreciated.

Reply:
You would either need to overload it per label_type or pass the label_type as a parameter, but yes a central place to check this would be nice

Reply:
I moved the logic to a function in learner.h which takes label_type as an input. Let me know if there's a better place to move this function.